passive/detect_samesite_protection.js

```diff
@@ -0,0 +1,65 @@
+/*
+Script to detect if the site use the protection bring by the "SameSite" cookie attribute.
+
+Knowing that point is interesting because the goal of this attribute is to mitigate CSRF attack.
+
+Links:
+- https://chloe.re/2016/04/13/goodbye-csrf-samesite-to-the-rescue
+- https://tools.ietf.org/html/draft-west-first-party-cookies
+- https://www.chromestatus.com/feature/4672634709082112
+
+Author: dominique.righetto@gmail.com
+*/
+
+function scan(ps, msg, src) {
+	//Docs on alert raising function:
+	// raiseAlert(risk, int confidence, String name, String description, String uri, 
+	//		String param, String attack, String otherInfo, String solution, String evidence, 
+	//		int cweId, int wascId, HttpMessage msg)
+	// risk: 0: info, 1: low, 2: medium, 3: high
+	// confidence: 0: falsePositive, 1: low, 2: medium, 3: high, 4: confirmed
+
+	//Common variables
+	var cweId = 352; 
+	var wascId = 9;
+	var url = msg.getRequestHeader().getURI().toString();	
+	var cookieHeaderNames = ["Set-Cookie", "Set-Cookie2"];
+	var cookieSameSiteAttributeNameLower = "samesite";
+
+	//Response headers collection
+	var responseHeaders = msg.getResponseHeader();
+
+	//Detect and analyze presence of the cookie headers
+	for(var i = 0 ; i < cookieHeaderNames.length ; i++){
+		var headerName = cookieHeaderNames[i];
+		if(responseHeaders.getHeaders(headerName)){
+			//Check if the cookie header values contains the SameSite attribute
+			var headerValues = responseHeaders.getHeaders(headerName).toArray();
+			for(var j = 0 ; j < headerValues.length ; j++){
+				var cookieValue = headerValues[j].toLowerCase();
+				var cookieAttributes = cookieValue.split(";");
+				//Inspect each attribute in order to avoid false-positive spot
+				//by simply searching "samesite=" on the whole cookie header value...
+				var sameSiteAttrFound = false;
+				var sameSiteAttrValue = null;
+				for(var k = 0 ; k < cookieAttributes.length ; k++){
+					var parts = cookieAttributes[k].split("=");
+					if(parts[0].trim() === cookieSameSiteAttributeNameLower){
```

Comment:
This yields always `false` with Java 7 (Rhino engine), does it really need to be `===`?